### PR TITLE
Add Python binding `resizable` to class `{Untyped,Typed}Storage`

### LINKED ISSUE #119233: Add a binding for Storage resizable
### 🐛 Describe the bug

Internal xref: https://fb.workplace.com/groups/1405155842844877/posts/7973507386009657/

StorageImpl has this field

```
  bool resizable_;
```

but there is no way to access it from Python. Make it accessible under a method named resizable() for both TypedStorage and UntypedStorage.

### Versions

main

cc @albanD

### PR DESCRIPTION
**Pull request by hkmatsumoto**
This PR exposes `resizable` method of `StorageImpl` to Python frontend to make it accessible for users.

Fixes #119233

### CODE REVIEW

File: torch/csrc/StorageMethods.cpp [hunk 1]
```diff
@@ -62,6 +62,13 @@ static PyObject* THPStorage_dataPtr(PyObject* self, PyObject* noargs) {
   END_HANDLE_TH_ERRORS
 }
 
+static PyObject* THPStorage_resizable(PyObject* self, PyObject* noargs) {
+  HANDLE_TH_ERRORS
+  THPStorage_assertNotNull(self);
+  return PyBool_FromLong(THPStorage_Unpack(self).resizable());
```
Comment (mikaylagawarecki) [inline torch/csrc/StorageMethods.cpp:68]: Why `PyBool_FromLong`, (might have missed something but doesn't `.resizable()` return `bool`?)

https://github.com/pytorch/pytorch/blob/238d87f74df7cbf38ce55c89abd18c5234ee55c7/c10/core/Storage.h#L101-L103

**Review by mikaylagawarecki — COMMENTED** [review]
Looks ok other than one comment. 


fyi @ezyang since you created the issue

File: torch/csrc/StorageMethods.cpp (hunk 1 above)
Comment (ezyang) [reply torch/csrc/StorageMethods.cpp:68]: I mean, there's no PyBool_FromBool lol, cuz bool isn't a thing in C haha.

**Review by ezyang — APPROVED** [review]

File: torch/csrc/StorageMethods.cpp (hunk 1 above)
Comment (mikaylagawarecki) [reply torch/csrc/StorageMethods.cpp:68]: oh whoops, got it!

File: test/test_torch.py [hunk 2]
```diff
@@ -8305,6 +8305,12 @@ def test_sizeof(self) -> None:
         self.assertEqual((sizeof_100 - sizeof_empty) // (sizeof_10 - sizeof_empty), 10)
         self.assertEqual((sizeof_100 - sizeof_empty) % (sizeof_10 - sizeof_empty), 0)
 
+    def test_resizable(self) -> None:
```
Comment (mikaylagawarecki) [inline test/test_torch.py:8308]: This test might need to be decorated with @skipIfTorchDynamo() looks like it [fails on the dynamo shard](https://hud.pytorch.org/pr/119286)

Comment (hkmatsumoto) [reply test/test_torch.py:8308]: Reproduced this behavior with `@torch.compile` on local machine so `@skipIfTorchDynamo` should be the way to go.
For my info, when JIT compiled, does `x.numpy()` allocates another chunk of memory with the content of `x`?

Comment (ezyang) [reply test/test_torch.py:8308]: When JIT compiled, we actually transparently avoid the conversion to numpy entirely. So that's why this happens.

Comment (hkmatsumoto) [reply test/test_torch.py:8309]: Understood, thanks!

**Comment by hkmatsumoto** [comment]
Force-pushed as I messed up the commit message

**Review by mikaylagawarecki — APPROVED** [review]

**Pull request closed without merging**